src/main/java/com/mapzen/valhalla/Instruction.kt

```diff
@@ -0,0 +1,202 @@
+package com.mapzen.valhalla
+
+import com.mapzen.helpers.DistanceFormatter
+import com.mapzen.ontheroad.R
+
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+
+import android.content.Context
+import android.location.Location
+import android.util.Log
+
+import java.util.Locale
+
+import com.mapzen.osrm.Route.SNAP_PROVIDER
+
+public class Instruction {
+    public val NONE : Int = 0
+    public val START : Int = 1
+    public val START_RIGHT : Int = 2
+    public val START_LEFT : Int = 3
+    public val DESTINATION : Int = 4
+    public val DESTINATION_RIGHT : Int = 5
+    public val DESTINATION_LEFT : Int = 6
+    public val BECOMES : Int = 7
+    public val CONTINUE : Int = 8
+    public val SLIGHT_RIGHT : Int = 9
+    public val RIGHT : Int = 10
+    public val SHARP_RIGHT : Int = 11
+    public val U_TURN_RIGHT: Int = 12
+    public val U_TURN_LEFT : Int = 13
+    public val SHARP_LEFT : Int = 14
+    public val LEFT : Int = 15
+    public val SLIGHT_LEFT : Int = 16
+    public val RAMP_STRAIGHT : Int = 17
+    public val RAMP_RIGHT : Int = 18
+    public val RAMP_LEFT : Int = 19
+    public val EXIT_RIGHT : Int = 20
+    public val EXIT_LEFT : Int = 21
+    public val STAY_STRAIGHT : Int = 22
+    public val STAY_RIGHT : Int = 23
+    public val STAY_LEFT : Int = 24
+    public val MERGE : Int = 25
+    public val ROUNDABOUT_ENTER : Int = 26
+    public val ROUNDABOUT_EXIT: Int = 27
+    public val FERRY_ENTER: Int = 28
+    public val FERRY_EXIT: Int = 29
+
+    private var json: JSONObject? = null;
+
+    public var turnInstruction: Int = 0
+    public var distance: Int = 0
+    public var location: Location = Location(SNAP_PROVIDER)
+    public var liveDistanceToNext: Int = -1
+    public var bearing: Int = 0
+
+    throws(javaClass<JSONException>())
+    public constructor(json: JSONObject) {
+        if (json.length() < 6) {
+            throw JSONException("too few arguments")
+        }
+        this.json = json
+        turnInstruction = parseTurnInstruction(json)
+        distance = (json.getDouble("length") * 1000).toInt()
+    }
+
+    /**
+     * Used for testing. Do not remove.
+     */
+    SuppressWarnings("unused")
+    protected constructor() {
+    }
+
+    public fun getIntegerInstruction(context: Context): Int {
+       return turnInstruction;
+    }
+
+    public fun getHumanTurnInstruction(context: Context): String {
+        return json!!.getString("instruction");
```

Comment:
Be careful with the `!!` operator. This is the best way to get NPEs in Kotlin (if you really want them you can have them). Probably better to have the return type be `String?`.